pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/DeleteOncHistoryPatch.java

```diff
@@ -0,0 +1,21 @@
+package org.broadinstitute.dsm.model.patch;
+
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.dsm.util.NotificationUtil;
+
+@Slf4j
+public class DeleteOncHistoryPatch extends ExistingOncHistoryPatch {
+    private final NotificationUtil notificationUtil;
+    public DeleteOncHistoryPatch(Patch patch, NotificationUtil notificationUtil) {
```

Comment:
So originally the delete patch requests were considered an existing patch, so I made these two classes `DeleteOncHistoryPatch` and `DeleteTissuePatch` that are extended from the original class and the only thing I changed is that I added one extra step in their `doPatch` method to call `DeletePatchFactory.setDeletedForChildrenFields(this.patch, notificationUtil);` I have explained that more in its class